backend/src/webhook/webhook.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@nestjs/common';
+import { Availability } from 'src/availability/entities/availability.entity';
+import { Planning } from 'src/planning/entities/planning.entity';
+import dayjs from 'dayjs';
+
+@Injectable()
+export class WebhookService {
+
+  sendWebhook(planning: Planning, availabilities: Availability[]) {
+    const maxAvailability = this.getMaxAvailability(planning, availabilities)
+
+    let daysText = ""
+    for(const day of maxAvailability.days){
+      if(!day.times.length) continue;
+      daysText += `${day.day}: ${day.times.map(time=>this.getTimeRangeString(time)).join(' ')}\n`
+    }
+
+    fetch(planning.webhook, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify({
+        content: `[Voted](${process.env.VITE_FRONTEND_URL}/${planning.id}): ${maxAvailability.voted.join(', ')}\n${daysText}`
+      })
+    })
+  }
+
+  getMaxAvailability(planning: Planning, availabilities: Availability[]) {
+    const days: {
+      day: string,
+      times: {
+        startTime: number,
+        endTime: number
+      }[]
+    }[] = []
+
+    const times = availabilities.flatMap(availability=>availability.times)
+    const userAmount = availabilities.length
+    
+
+    let currentDate = dayjs.unix(planning.startDate)
+    const endDate   = dayjs.unix(planning.endDate)
+    while(currentDate.isBefore(endDate, 'day') || currentDate.isSame(endDate, 'day')){
+      const currentDayTimes = times.filter(time=>currentDate.isSame(dayjs.unix(time.startTime), 'day'))
+
+      const maxAvailabilityTimeBlocks: {
+        startTime: number,
+        endTime: number
+      }[] = []
+      const processedTimes: number[] = []
+      for(const timeBlock of currentDayTimes){
+        const startTime = timeBlock.startTime
+        if(processedTimes.includes(startTime)) continue;
+        processedTimes.push(startTime)
+        const matchingBlocks = currentDayTimes.filter(block=>
```

Comment:
For example with this you know the smallest end time with only 1 loop:
let smallestEndTime = Number.POSITIVE_INFINITY (above the for loop)

```suggestion
const matchingBlocks = currentDayTimes.filter(block=> {
          smallestEndTime = block.endTime < smallestEndTime ? block.endTime : smallestEndTime
          startTime >= block.startTime
          && startTime < block.endTime
        })
```